### ISSUE TITLE
[BUG] Rule is not triggered on `implements`

### ISSUE THREAD

**Issue by ibragins** [konveyor/analyzer-lsp#855]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Undefined (Default)

### Current Behavior

```
when:
  java.referenced:
    location: IMPLEMENTS_TYPE
    pattern: implements AnotherClass
```
Should trigger an incident in the following code (extracted from [tackle-testapp](https://github.com/konveyor/tackle-testapp-public/blob/main/src/main/java/io/konveyor/demo/ordermanagement/OrderManagementAppInitializer.java#L12))
```
public class OrderManagementAppInitializer implements WebApplicationInitializer  {
```
But it doesn't trigger any incidents.

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by abrugaro** [comment]
You're right @shawn-hurley, there was a mistake when reporting the bug.

@ibragins [this](https://github.com/abrugaro/test-rules/blob/main/java/java-basic-locations.yaml#L173) is the rule used in our tests and it is passing. I'll update the rule to remove the bug label.

This bug can be closed, thanks @shawn-hurley.
